session/session.go

```diff
@@ -437,6 +438,14 @@ func (s *sessionImpl) Bind(ctx context.Context, uid string) error {
 		if val, ok := s.pool.sessionsByUID.Load(uid); ok {
 			val.(Session).Close()
 		}
+		// although here are checking old session and invoking Close method,
+		// there is a possibility that the method is invoked from HandlerService.Handle's defer logic at the same time from different goroutine
+		// so we need to make sure two things:
+		// 1. ensure the Close method's logic is done only once, for the reason to prevent deleting the new one
+		// 2. double check that there isn't a session in the store data before Storing
+		for s.pool.GetSessionByUID(uid) != nil {
```

Comment:
Can we put a timeout here so that this doesn't turn into a dead lock in corner cases?

Reply:
Sure, That's great. But I am not sure that how much timeout value we could set, is 30s enough? Or set the loop counter limit  directly?

Reply:
Make it configurable